external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -625,4 +620,21 @@ public String toString() {
     private String getTopicsString() {
         return kafkaSpoutConfig.getSubscription().getTopicsString();
     }
+    
+    private static class PollablePartitionInfo {
+        private final Set<TopicPartition> pollablePartitions;
+        //The subset of earliest retriable offsets that are on pollable partitions
+        private final Map<TopicPartition, Long> pollableEarliestRetriableOffsets;
+        
+        public PollablePartitionInfo(Set<TopicPartition> pollablePartitions, Map<TopicPartition, Long> earliestRetriableOffsets) {
+            this.pollablePartitions = pollablePartitions;
+            this.pollableEarliestRetriableOffsets = earliestRetriableOffsets.entrySet().stream()
+                .filter(entry -> pollablePartitions.contains(entry.getKey()))
+                .collect(Collectors.toMap(entry -> entry.getKey(), entry -> entry.getValue()));
+        }
+        
+        public boolean isPollAllowed() {
```

Comment:
why not simply `isEmpty()` ? 

I really don't thing that _allowed_ is a very suitable name here because we are not checking for permissions or anything like that. If it is in the pollable set it is always allowed.

Reply:
I agree that allowed can be misunderstood. I'd prefer not calling it isEmpty, since this isn't a collection and it's not obvious which internal collection it would be referring to. How about shouldPoll?

Reply:
Building on the comment above, I would still call it something along the lines `isReadyToPoll()` or `isPollableRetryReady`, because that's what it is, a retry that is ready to poll. However, it's up to you and I am +1 after this.